web/sse.go

```diff
@@ -63,10 +74,17 @@ func (s *SseServer) Price(pattern string) {
 			prices := s.priceDispatcher.Subscribe(symbol)
 			for {
 				select {
+				case <-ctx.Done():
+					log.Println("Context done")
+					return
 				case <-timeout:
 					log.Println("Timeout")
 					s.priceDispatcher.Unsubscribe(symbol)
 					return
+				case <-r.Context().Done():
+					log.Println("Context done")
+					s.priceDispatcher.Unsubscribe(symbol)
+					return
 				case price := <-prices:
```

Comment:
You are still not handling the case when another thread closing the same "prices" channel.
If another goroutine close the channel and this one is not timed out, then you will send the default value rapidly and maybe endlessly to the client in the "<-prices" case.

But this concept is wrong, see my comment in the dispatcher process method.